pyproject.toml

```diff
@@ -16,90 +22,90 @@ keywords = ["MLOps", "AI", "Model Serving", "Model Deployment", "Machine Learnin
 "Baseten" = "https://baseten.co"
 
 [tool.poetry.dependencies]
-python = ">=3.8,<3.12"
-numpy = ">=1.23.5"
-msgpack = ">=1.0.2"
-msgpack-numpy = ">=0.4.7.1"
-packaging = ">=20.9"
-python-json-logger = ">=2.0.2"
-PyYAML = "^6.0"
-Jinja2 = "^3.1.2"
-python-on-whales = "^0.68.0"
-tenacity = "^8.0.1"
-single-source = "^0.3.0"
 blake3 = "^0.3.3"
+boto3 = "^1.26.157"
 fastapi = "^0.109.1"
-uvicorn = "^0.24.0"
+google-cloud-storage = "2.10.0"
```

Comment:
How much of these are required to be pinned to exact versions and how much can we loosen these?

Reply:
That's a good question.

We could potentially convert all version specifiers to a minimum version, where the minimum is the oldest we want to allow/support customers to use, while generally trying to resolve the versions to the newest that is compatible.

Though I don't want to touch any global version changes in this PR without having better understanding of the pros and cons of doing that and implications. 

The changes here keep all versions (except pydantic) and just sort the deps.

Reply:
Good question. Probably we could change (some?) to minimal versions instead of exact. But I don't want to change any versions globally, except the pydantic version, in this PR. I'm not sure what the implications would be.
All changes here are just from sorting the deps, not changing the versions.